core/templates/dev/head/pages/exploration_editor/editor_tab/StateInteraction.js

```diff
@@ -287,6 +308,48 @@ oppia.controller('StateInteraction', [
                 stateCustomizationArgsService.displayed = {};
               };
 
+              $scope.getSaveInteractionButtonStyleTooltip = function() {
+                if (!$scope.hasCustomizationArgs) {
+                  return 'No customization arguments';
+                }
+                if (!$scope.stateInteractionIdService.displayed) {
+                  return 'No interaction being displayed';
+                }
+
+                var warningsList = $scope.getCustomizationArgsWarningsList();
+                var warningMessages = warningsList.map(function(warning) {
+                  return warning.message;
+                });
+
+                if (warningMessages.length === 0) {
+                  if ($scope.form.schemaForm.$invalid) {
+                    return 'Some of the form controls failed validation';
+                  } else {
+                    return '';
+                  }
+                } else if (warningMessages.length === 1) {
+                  return warningMessages[0];
+                } else {
+                  return '• ' + warningMessages.join('\n• ');
+                }
+              }
+
+              $scope.getSaveInteractionButtonStyle = function() {
+                if ($scope.hasCustomizationArgs &&
+                    $scope.stateInteractionIdService.displayed &&
+                    $scope.form.schemaForm.$valid &&
+                    $scope.areCustomizationArgsValid()) {
+                  return '';
+                } else {
+                  return 'background-size: 20px 20px; ' +
+                    'background-position: 6px; ' +
+                    'padding-left: 32px; ' +
+                    'background-repeat: no-repeat; ' +
+                    'background-image: ' +
+                    'url("../assets/common/images/general/warning.png")';
```

Comment:
I played with this for a long time on Sunday and some of these functions were being called even when some of the pre-requisites weren't met, despite the ng-if of the elements involved, so I got a bit paranoid and started checking all these conditions everywhere.

Nevertheless, I cleaned this up a bit now. Let me know what you think.